lib/irb.rb

```diff
@@ -705,6 +735,18 @@ def inspect
       format("#<%s: %s>", self.class, ary.join(", "))
     end
 
+    def assignment_expression?(line)
+      # Try to parse the line and check if the last of possibly multiple
+      # expressions is an assignment type.
+
+      # If the expression is invalid, Ripper.sexp should return nil which will
+      # result in false being returned. Any valid expression should return an
+      # s-expression where the second selement of the top level array is an
+      # array of parsed expressions. The first element of each expression is the
+      # expression's type.
+      ASSIGNMENT_NODE_TYPES.include?(Ripper.sexp(line)&.dig(1,-1,0))
```

Comment:
Please use `Ripper.lex` because `Ripper.sexp` is not maintained.

Reply:
I don't think that `Ripper.sexp` is less maintained than `Ripper.lex`, and vice versa.